src/checks/checkNodeVersion.js

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License"); 
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS 
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+// const array specifies accepted versions of NodeJS
+// can specify major/minor versions, ex: 14, 14.13, 16.34.02
+const VERSIONS = [14, 16]
+
+async function checkNodeVersion(metadataJSON) {
+    if (!metadataJSON) {
+        return {
+            message: 'Error: metadataJSON is null',
+            status: 'fail',
```

Comment:
Status "fail" isn't the best option in this case. In general "fail" isn't a good word, it doesn't follow any practice and can be applied only in UI but not programmatically. In this case, I propose to use boolean value and instead of status property, use "error".  
So, the code should look like: 
```
return {
    message: 'Error: metadataJSON is null',
    error: true
}
```
A best practice if you needed to introduce some special words, like 'fail', 'pass' etc is to introduce constants that will be sharable between modules, and in most cases is good to represent them as enum type. This will give the possibility to not depend exactly on the string, we will depend on the entity.  It should be something like:
```
module.exports = {
    FAIL: 0,
    PASS: 1,
    SOME_OTHER_STATUS: 2
    ...
}
```

but again, we need to try to avoid it were it possible. If we have just 2 statuses the better choice use the boolean value.


For errors, you can use the new Error('Message').toString() construction if you want have error as a string. For simplifying debugging you also could add stack trace by new Error('Message').stack call. 

if (result.status === 'pass') {
                    return true
                }

Reply:
I'll think about this as a possible area to refactor. I get your point about the statuses, and I'll probably change them into boolean/enum at some point. However, on the subject of errors, I don't think checks failing should throw an error for the entire test suite, which is why I lumped bad inputs and missing keys into the 'fail' status alongside the check condition.